packages/features/data-table/components/segment/SaveFilterSegmentButton.tsx

```diff
@@ -0,0 +1,223 @@
+import { useSession } from "next-auth/react";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { MembershipRole, type FilterSegmentScope } from "@calcom/prisma/enums";
+import { trpc } from "@calcom/trpc/react";
+import { Button } from "@calcom/ui/components/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTrigger,
+} from "@calcom/ui/components/dialog";
+import { Form, Input, Label, Select, Switch } from "@calcom/ui/components/form";
+import { RadioGroup, RadioField } from "@calcom/ui/components/radio";
+import { showToast } from "@calcom/ui/components/toast";
+
+import { useDataTable } from "../../hooks";
+
+interface FormValues {
+  name: string;
+  scope: FilterSegmentScope;
+  teamId?: number;
+}
+
+export function SaveFilterSegmentButton() {
+  const { t } = useLocale();
+  const utils = trpc.useUtils();
+  const [isOpen, setIsOpen] = useState(false);
+  const [isTeamSegment, setIsTeamSegment] = useState(false);
+  const [selectedTeamId, setSelectedTeamId] = useState<number>();
+  const session = useSession();
+  const orgRole = session?.data?.user?.org?.role;
+  const isAdminOrOwner = orgRole === MembershipRole.OWNER || orgRole === MembershipRole.ADMIN;
```

Comment:
NIT: This is duplicated a lot - i wonder if we should have a useOrgAdminOrOwner hook to get this. Not sure if that exists already but might be a nice change to add in a follow up deffo not blocking

Reply:
Yeah auth might be a good place for it! - Not blocking by any means tho

Reply:
yeah i'll do it in a separate PR :D